test/sqllogictest/dates-times.slt

```diff
@@ -1368,3 +1368,106 @@ query T
 SELECT now() + null;
 ----
 NULL
+
+# Test timestamp precision.
+
+query T
+SELECT '1970-01-01T00:00:00.666666'::timestamp(0);
+----
+1970-01-01 00:00:01
+
+query T
+SELECT '1970-01-01T00:00:00.666666'::timestamp(1);
+----
+1970-01-01 00:00:00.7
+
+query T
+SELECT '1970-01-01T00:00:00.666666'::timestamp(2);
+----
+1970-01-01 00:00:00.67
+
+query T
+SELECT '1970-01-01T00:00:00.666666'::timestamp(3);
+----
+1970-01-01 00:00:00.667
+
+query T
+SELECT '1970-01-01T00:00:00.666666'::timestamp(4);
+----
+1970-01-01 00:00:00.6667
+
+query T
+SELECT '1970-01-01T00:00:00.666666'::timestamp(5);
+----
+1970-01-01 00:00:00.66667
+
+query T
+SELECT '1970-01-01T00:00:00.666666'::timestamp(6);
+----
+1970-01-01 00:00:00.666666
+
+
+query T
+SELECT '1970-01-01T00:00:00.666666'::timestamptz(0);
+----
+1970-01-01 00:00:01+00
+
+query T
+SELECT '1970-01-01T00:00:00.666666'::timestamptz(1);
+----
+1970-01-01 00:00:00.7+00
+
+query T
+SELECT '1970-01-01T00:00:00.666666'::timestamptz(2);
+----
+1970-01-01 00:00:00.67+00
+
+query T
+SELECT '1970-01-01T00:00:00.666666'::timestamptz(3);
+----
+1970-01-01 00:00:00.667+00
+
+query T
+SELECT '1970-01-01T00:00:00.666666'::timestamptz(4);
+----
+1970-01-01 00:00:00.6667+00
+
+query T
+SELECT '1970-01-01T00:00:00.666666'::timestamptz(5);
+----
+1970-01-01 00:00:00.66667+00
+
+query T
+SELECT '1970-01-01T00:00:00.666666'::timestamptz(6);
+----
+1970-01-01 00:00:00.666666+00
+
+query T
+SELECT '1970-01-01T00:00:00.666666'::timestamp::timestamp(3);
+----
+1970-01-01 00:00:00.667
+
+query T
+SELECT '1970-01-01T00:00:00.666666'::timestamptz::timestamptz(3);
+----
+1970-01-01 00:00:00.667+00
+
+query T
+SELECT '1970-01-01T00:00:00.666666'::timestamp::timestamptz(3);
+----
+1970-01-01 00:00:00.667+00
+
+query T
+SELECT '1970-01-01T00:00:00.666666'::timestamptz::timestamp(3);
+----
+1970-01-01 00:00:00.667
+
+query T
+SELECT '1970-01-01T00:00:00.666666'::date::timestamptz(3);
+----
+1970-01-01 00:00:00+00
+
+query T
+SELECT '1970-01-01T00:00:00.666666'::date::timestamp(3);
+----
+1970-01-01 00:00:00
```

Comment:
Could also use a battery of tests that ensure that timestamps with precisions interoperate, e.g.

```
postgres=# SELECT '1970-01-01T00:00:00.666666'::timestamp(5) - '1970-01-01T01:00:00.6666'::timestamp(3);
    ?column?     
-----------------
 -01:00:00.00033
```

`date_bin`, `age` all seem like good candidates

Reply:
Ack, let me add those.

Reply:
Added!